test/acceptance/test/core_operations_test.go

```diff
@@ -64,176 +65,314 @@ var (
 	client gitprovider.Client
 )
 
-// Run core operations and check status
-func TestCoreOperations(t *testing.T) {
-	tmpPath, err := ioutil.TempDir("", "tmp-dir")
-	log.Infof("Using temp directory: %s", tmpPath)
+var _ = Describe("WEGO Acceptance Tests", func() {
 
-	require.NoError(t, err)
-	defer os.RemoveAll(tmpPath)
-	tmpDir = tmpPath
+	var session *gexec.Session
+	var err error
+	var tmpPath string
 
-	log.Info("Creating GitHub client...")
-	token, found := os.LookupEnv("GITHUB_TOKEN")
-	require.True(t, found)
-	c, err := github.NewClient(github.WithOAuth2Token(token), github.WithDestructiveAPICalls(true), github.WithConditionalRequests(true))
-	require.NoError(t, err)
-	client = c
-	log.Info("Ensuring wego repo does not exist...")
-	ensureWegoRepoIsAbsent(t)
-	log.Info("Ensuring flux version is set...")
-	ensureFluxVersion(t)
-	log.Info("Checking initial status...")
-	checkInitialStatus(t)
-	log.Info("Install flux...")
-	installFlux(t)
-	log.Info("Setting up test repository...")
-	setUpTestRepo(t) // create repo with simple nginx manifest
-	defer deleteRepos(t)
-	log.Info("Adding test repository to cluster...")
-	require.NoError(t, err)
-	addRepo(t) // add new repo to cluster
-	log.Info("Waiting for workload to start...")
-	waitForNginxDeployment(t)
-}
+	AfterEach(func() {
+		os.RemoveAll(tmpPath)
+		Expect(deleteRepos()).Should(Succeed())
+	})
 
-func addRepo(t *testing.T) {
-	keyFilePath := filepath.Join(os.Getenv("HOME"), ".ssh", "id_rsa")
-	if _, err := os.Stat(keyFilePath); os.IsNotExist(err) {
-		key := os.Getenv("GITHUB_KEY")
-		tmpFile, err := ioutil.TempFile("", "keyfile")
-		require.NoError(t, err)
-		defer tmpFile.Close()
-		require.NoError(t, ioutil.WriteFile(tmpFile.Name(), []byte(key), 600))
-		keyFilePath = tmpFile.Name()
-	}
+	BeforeEach(func() {
+		By("Given I have a wego binary installed on my local machine", func() {
+			Expect(FileExists(WEGO_BIN_PATH)).To(BeTrue())
+		})
 
-	dir, err := os.Getwd()
-	require.NoError(t, err)
-	require.NoError(t, os.Chdir(tmpDir))
-	defer func() {
-		require.NoError(t, os.Chdir(dir))
-	}()
+		By("Setup test", func() {
+			Expect(checkInitialStatus()).Should(Succeed())
+			Expect(setupTest()).Should(Succeed())
+			Expect(ensureWegoRepoIsAbsent()).Should(Succeed())
+			Expect(ensureFluxVersion()).Should(Succeed())
+			Expect(installFlux()).Should(Succeed())
+			Expect(setUpTestRepo()).Should(Succeed())
+		})
+
+	})
+
+	It("Verify add private repo when repo does not already exist", func() {
+
+		By("When i run 'wego add .'", func() {
+			dir, err := os.Getwd()
+			Expect(err).Should(Succeed())
+			Expect(os.Chdir(tmpDir)).Should(Succeed())
+			Expect(os.Chdir(dir)).Should(Succeed())
```

Comment:
Why are we changing to do different directories in a row? Is it to test if `tmpDir` is set up correctly?

Reply:
I actually just copied that over from yours. I wasnt really sure.

Reply:
Do you mean this?
```
	require.NoError(t, os.Chdir(tmpDir))
	defer func() {
		require.NoError(t, os.Chdir(dir))
	}()
```
That's different... it's changing back at the end of the function due to the defer

Reply:
Yeah i do see that